### DISCUSSION TITLE
I want to modify the duration count.

### DISCUSSION THREAD

**Discussion by zhasan0** [kimai/kimai#2407] [General]
There have only begin and end time. But I added extra two fields as a afternoon shift begin and end time. Duration will count based on begin and end time.  But now how can I modify the duration counting. I want to the duration count like below image.

![SCREEN-1](https://user-images.githubusercontent.com/42202246/110311178-9fee2480-802d-11eb-990b-b4277ffeba09.png)

**Comment by kevinpapst** [comment]
You mean this?
https://github.com/kevinpapst/kimai2/blob/master/src/Timesheet/Calculator/DurationCalculator.php

**Reply by zhasan0** [reply]
Hello great people, I am a beginner developer. Who wants to learn more. Can you please help me to solve this problem.

I modified the DurationCalculator file like below image. But it's do not working like new modification. It's work like existing duration count.

Thank you.
![duration calculaton modified](https://user-images.githubusercontent.com/42202246/110346919-3fc1a780-805a-11eb-823b-820bbb040145.png)
